include/rcpputils/filesystem_helper.hpp

```diff
@@ -255,10 +255,31 @@ RCPPUTILS_PUBLIC bool exists(const path & path_to_check);
 /**
  * \brief Get a path to a location in the temporary directory, if it's available.
  *
+ * This does not create any directories.
+ * On Windows, this uses "GetTempPathA"
+ * On non-Windows, this prefers the environment variable TMPDIR, falling back to /tmp
+ *
  * \return A path to a directory for storing temporary files and directories.
  */
 RCPPUTILS_PUBLIC path temp_directory_path();
 
+/**
+ * \brief Construct a uniquely named temporary directory, in "parent", with format base_name.XXXXXX
+ *
+ * The output, if successful, is guaranteed to be a newly-created directory.
+ * The underlying utilities check if the generated path name exists until a one is found.
```

Comment:
@emersonknapp did you mean:
```suggestion
 * The underlying implementation keeps generating paths until one that does not exists is found.
```